src/subcommand/server/accept_encoding.rs

```diff
@@ -0,0 +1,123 @@
+use {super::*, axum::extract::FromRef};
+
+pub(crate) struct AcceptEncoding(pub(crate) Vec<String>);
+
+#[async_trait::async_trait]
+impl<S> axum::extract::FromRequestParts<S> for AcceptEncoding
```

Comment:
Let's change this to return the header value as a `String`, to avoid allocation a `Vec<String>`, and then we can do the check in the content response function. Content repsonse should take a `AcceptEncoding` so we don't have a random `String` arg